helper/impls/helper-impl-110/src/main/java/com/linkedin/avroutil1/compatibility/avro110/FieldBuilder110.java

```diff
@@ -88,12 +97,57 @@ public FieldBuilder copyFromField() {
 
   @Override
   public Schema.Field build() {
-    Schema.Field result = new Schema.Field(_name, _schema, _doc, _defaultVal, _order);
+    Object avroFriendlyDefault;
+    try {
+      avroFriendlyDefault = avroFriendlyDefaultValue(_defaultVal);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("unable to convert default value " + _defaultVal + " into something avro can handle", e);
+    }
+    Schema.Field result = new Schema.Field(_name, _schema, _doc, avroFriendlyDefault, _order);
     if (_props != null) {
       for (Map.Entry<String, Object> entry : _props.entrySet()) {
         result.addProp(entry.getKey(), entry.getValue());
       }
     }
     return result;
   }
+
+  /**
+   * we want to be very generous with what we let users provide for default values.
+   * sadly, avro can only handle specific classes/collections/primitive-wrappers
+   * (see {@link org.apache.avro.util.internal.JacksonUtils#toJson(Object, JsonGenerator)})
+   * @param mightNotBeFriendly a proposed field default value that might originate from
+   *                           a call like AvroCompatibilityHelper.getGenericDefaultValue()
+   * @return a representation of the input that avro likes for use as a field default value
+   */
+  private Object avroFriendlyDefaultValue(Object mightNotBeFriendly) throws Exception {
```

Comment:
Put this into Jackson2Utils.java so that it can be shared with Avro 1.9 and reduce code duplication.